src/key.rs

```diff
@@ -395,11 +395,15 @@ impl PublicKey {
 
     /// Adds the keys in the provided slice together, returning the sum. Returns
     /// an error if the result would be the point at infinity, i.e. we are adding
-    /// a point to its own negation
+    /// a point to its own negation, or if the provided slice has no element in it.
     pub fn combine_keys(keys: &[&PublicKey]) -> Result<PublicKey, Error> {
         use core::mem::transmute;
         use core::i32::MAX;
 
+        if keys.len() < 1 {
+            return Err(InvalidPublicKeySum);
+        }
+
```

Comment:
Maybe it should be `!keys.is_empty() && keys.len() < i32::MAX as usize`? and then we can also remove the `debug_assert`